source/PlayerInfo.cpp

```diff
@@ -1353,33 +1363,22 @@ void PlayerInfo::Land(UI *ui)
 // which case a message will be returned.
 bool PlayerInfo::TakeOff(UI *ui)
 {
-	// This can only be done while landed.
-	if(!system || !planet)
+	if(!LeavePlanet())
 		return false;
 
 	flagship = FlagshipPtr();
 	if(!flagship)
 		return false;
 
-	shouldLaunch = false;
-	Audio::Play(Audio::Get("takeoff"));
-
-	// Jobs are only available when you are landed.
-	availableJobs.clear();
-	availableMissions.clear();
-	doneMissions.clear();
-	stock.clear();
+	// Move the flagship to the start of the list of ships and ensure that all
+	// escorts know which ship is acting as flagship.
+	SetFlagship(*flagship);
 
 	// Special persons who appeared last time you left the planet, can appear again.
 	GameData::ResetPersons();
 
-	// Store the total cargo counts in case we need to adjust cost bases below.
```

Comment:
Doesn't seem like this comment should be removed.

Reply:
fixed